src/coreclr/jit/emitxarch.cpp

```diff
@@ -6822,7 +6823,9 @@ void emitter::emitIns_R_R_A(
     id->idIns(ins);
     id->idReg1(reg1);
     id->idReg2(reg2);
+
     SetEvexBroadcastIfNeeded(id, instOptions);
+    SetEvexEmbMaskIfNeeded(id, instOptions);
```

Comment:
are there various formats for `Avx512F.Add` because of which you have to add `SetEvexEmbMaskIfNeeded` in various `emitIns*` methods?

Reply:
Yes, there is both register and memory based encodings, so we have to handle `R_R_R` (register) and `R_R_A`/`R_R_C`/`R_R_S` (memory).